### DISCUSSION TITLE
If theme.LightTheme() is deprecated

### DISCUSSION THREAD

**Discussion by bq-wrongway** [fyne-io/fyne#3575] [Q&A] [answered]
How to achieve for fyne app to change theme based on the underling system theme.

i was doing something like this 

```
if !isDark(darkMode) {
		a.Settings().SetTheme(theme.LightTheme())
	}
```

But i get a warning that the this will be deprecated in 3.0. Is there an OOB solution to autochaging dark/light based on system default?

**Comment by Jacalz** [comment, marked as answer]
We already do switch between light/dark theme automatically on most platforms. Version v2.3.0 added support for the feature on Linux (as long as the desktop supports the FreeDesktop Dark Theme Preference). We already had support for macOS/Windows before as far as I know. Is there any platform on which it doesn’t work for you?

For changing these settings, I personally used https://pkg.go.dev/fyne.io/fyne/v2/cmd/fyne_settings/settings#Settings.LoadAppearanceScreen in Rymdport (https://github.com/Jacalz/rymdport/blob/563f716f967de28830799cafd91d5c51934915c3/internal/ui/settings.go#L199) to integrate part of the fyne_settings application. I am not entirely sure that I would recommend it in all cases though because the downside is that it will change the settings for all Fyne applications. For containerised distribution (Flatpak, Snap and such) it is not problem because it can’t access the rest of the file system.

**Reply by Jacalz** [reply]
To be fair, if the automatic dark/light theme switching existed on Linux back then (the protocol is quite new), I likely never would have integrated the settings in that way. I mostly keep it as some users have expressed their appreciation for being able to change the scale within the app.

**Reply by Jacalz** [reply]
For some additional context, the API was deprecated as it should be up to the user (and their system) to change theme, not the developer forcing it on them.

**Reply by bq-wrongway** [reply]
Hey, thanks. You are right, my host machine is popOS, and since its based on gnome 40 it does not implement "proper" dark mode.
i tested on fedora 37 with gnome 42 and it works with no issues.

**Reply by andydotxyz** [reply]
You can always set `FYNE_THEME` or use the fyne_settings app to configure globally as well
